### PR TITLE
Use QML grabToImage to generate screenshot

### PR DESCRIPTION
**Pull request by piggz**
Fixes kernel oops on GS290 and GX290 aswell as fixes
a memory leak on @elros34's port

### CODE REVIEW

**Comment by piggz** [comment]
As 3 community devices use this patch, I wonder if its a generally safer method?

**Comment by piggz** [comment]
From elros, it wasnt a memory leak it fixed, it was blank screenshots.  Also may fix screenshots on the jolla tablet.

**Comment by pvuorela** [comment]
Wondering if this is just papering over some other bugs in those devices. By documentation the only limitation on grabWindow() appears to be: "Warning: This function can only be called from the GUI thread"

File: src/screenshotservice.cpp [hunk 1]
```diff
@@ -174,12 +176,15 @@ ScreenshotResult *ScreenshotService::saveScreenshot(const QString &path)
         return result;
     }
 
-    QImage grab(compositor->grabWindow());
+    auto grabResult = compositor->contentItem()->grabToImage();
```
Comment (pvuorela) [inline src/screenshotservice.cpp:179]: Could prefer the type written explicitly for code readability/understandability. Was first wondering why the qsharedpointer include when there's nothing needing it.

File: src/screenshotservice.h [hunk 2]
```diff
@@ -55,8 +55,9 @@ class LIPSTICK_EXPORT ScreenshotResult : public QObject
     Status m_status = Writing;
 };
 
-class ScreenshotService
+class ScreenshotService : public QObject
 {
+    Q_OBJECT
```
Comment (pvuorela) [inline src/screenshotservice.h:60]: Why the qobject?

**Comment by pvuorela** [comment]
No feedback.

**Comment by Thaodan** [comment]
> As 3 community devices use this patch, I wonder if its a generally safer method?

Care to comment?

**Comment by piggz** [comment]
Yes, give me a few days to find some time and i can make the changes, is it just the removal of q_object and use the correct type instead of auto?

**Comment by Thaodan** [comment]
> Yes, give me a few days to find some time and i can make the changes, is it just the removal of q_object and use the correct type instead of auto?

@pvuorela ^

**Comment by pvuorela** [comment]
> >  As 3 community devices use this patch, I wonder if its a generally safer method?
> Care to comment?

Think I already did. By documentation there shouldn't be any reason why not to use what is currently there and kernel oops sounds like a problem elsewhere.

**Pull request closed without merging**